communication/src/main/java/org/jboss/da/communcation/pnc/authentication/PNCAuthentication.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.jboss.da.communcation.pnc.ReadResource;
+
+/**
+ * Class obtained from pnc example on OAuth example on PNC:
+ * https://github.com/project-ncl/pnc/blob/master/examples/oauth-client/src/main/java/org/jboss/pnc/auth/client/SimpleOAuthConnect.java
+ */
+public class PNCAuthentication {
+
+    public static String getAccessToken(String url, Map<String, String> urlParams)
+            throws ClientProtocolException, IOException {
+        return connect(url, urlParams)[0];
+    }
+
+    public static String getRefreshToken(String url, Map<String, String> urlParams)
+            throws ClientProtocolException, IOException {
+        return connect(url, urlParams)[1];
+    }
+
+    public static String[] getTokens(String url, Map<String, String> urlParams)
+            throws ClientProtocolException, IOException {
+        return connect(url, urlParams);
+    }
+
+    public static String getAccessToken(String url, String clientId, String username, String password)
+            throws ClientProtocolException, IOException {
+        Map<String, String> urlParams = new HashMap<String, String>();
+        urlParams.put("grant_type", "password");
+        urlParams.put("client_id", clientId);
+        urlParams.put("username", username);
+        urlParams.put("password", password);
+        return connect(url, urlParams)[0];
+    }
+
+    public static String getrefreshToken(String url, String clientId, String username, String password)
+            throws ClientProtocolException, IOException {
+        Map<String, String> urlParams = new HashMap<String, String>();
+        urlParams.put("grant_type", "password");
+        urlParams.put("client_id", clientId);
+        urlParams.put("username", username);
+        urlParams.put("password", password);
+        return connect(url, urlParams)[1];
+    }
+
+    private static String[] connect(String url, Map<String, String> urlParams)
+            throws ClientProtocolException, IOException {
+
+        CloseableHttpClient httpclient = HttpClients.createDefault();
+        HttpPost httpPost = new HttpPost(url);
+
+        // add header
+        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded");
+
+        List<BasicNameValuePair> urlParameters = new ArrayList<BasicNameValuePair>();
+        for (String key : urlParams.keySet()) {
+            urlParameters.add(new BasicNameValuePair(key, urlParams.get(key)));
+        }
+        httpPost.setEntity(new UrlEncodedFormEntity(urlParameters));
+        CloseableHttpResponse response = httpclient.execute(httpPost);
+
+        String refreshToken = "";
+        String accessToken = "";
+        try {
```

Comment:
Weeeeeeeeeeee Java-7 stuff. I've been in Java-6 land for far too long :( will refactor!